lib/src/domain/user_cases/implementation/get_genres_use_case.dart

```diff
@@ -0,0 +1,17 @@
+import '../../../core/utils/state.dart';
+import '../../../data/datasource/remote/genre_api_provider.dart';
+import '../../../data/datasource/remote/i_api_service.dart';
+import '../../../data/repositories/genre_repository.dart';
+
+class GetGenresUseCase {
+  final ApiService api = GenreApiProvider();
+  late final GenreRepository genreRepo = GenreRepository(
+    api: api,
+  );
```

Comment:
You can directly use as a parameter the repository on the GetGenresUseCase like
```
GetGenresUseCase({
    required this.repository,
});

final IRepository repository;
```

Reply:
Changed!